app/gui2/src/workers/visualizationCompiler.ts

```diff
@@ -239,36 +273,52 @@ function compileError(path: string, error: Error) {
   postMessage<CompileError>({ type: 'compile-error', path, error })
 }
 
-async function tryFetch(path: string) {
+interface FetchResponse {
+  contents: ArrayBuffer
+  contentType: string | undefined
+}
+
+const fetchCallbacks = new Map<
+  string,
+  {
+    resolve: (contents: FetchResponse) => void
+    reject: (error: Error) => void
+  }
+>()
+
+function fetch_(path: string) {
+  return new Promise<FetchResponse>((resolve, reject) => {
+    fetchCallbacks.set(path, { resolve, reject })
+    postMessage<FetchWorkerRequest>({ type: 'fetch-worker-request', path })
+  })
+}
```

Comment:
It would be nicer to have a name without trailing `_` even if a bit redundant. Like `fetchResource`.